core/src/main/java/com/amplitude/core/Amplitude.kt

```diff
@@ -68,104 +73,131 @@ open class Amplitude internal constructor(
     }
 
     @Deprecated("Please use 'track' instead.", ReplaceWith("track"))
-    fun logEvent(event: BaseEvent) {
-        track(event)
+    fun logEvent(event: BaseEvent): Amplitude {
+        return track(event)
     }
 
-    fun track(event: BaseEvent, callback: EventCallBack? = null) {
+    /**
+     * Track an event
+     */
+    fun track(event: BaseEvent, callback: EventCallBack? = null): Amplitude {
         callback ?. let {
             event.callback = it
         }
         process(event)
+        return this
     }
 
-    fun track(eventType: String, eventProperties: JSONObject? = null, options: EventOptions? = null) {
+    /**
+     * Log event with the specified event type, event properties, and optional event options
```

Comment:
Log an event.

Q: Why not we fully leverage javadoc? Like using @param ?

Reply:
We can add param if needed, the only reason for not doing that earlier is https://kotlinlang.org/docs/coding-conventions.html#documentation-comments 

I can add these to account for java style, kdoc still supports them